### ISSUE TITLE
Support bitcoin RPC auth via "cookie" file

### ISSUE THREAD

**Issue by justinmoon** [fedimint/fedimint#1990]

**Comment by oleonardolima** [comment]
This one seems pretty straightforward, by using the [rust-bitcoincore-rpc `Auth::CookieFile`](https://docs.rs/bitcoincore-rpc/0.16.0/src/bitcoincore_rpc/client.rs.html#194) [here](https://github.com/fedimint/fedimint/blob/c5cb7dc811ac036cb976d4481471df3ec7491a7f/fedimint-bitcoind/src/bitcoincore_rpc.rs#L33-L59), is that right ?

I guess the problem would be how to get the path for the cookie file, a new environment variable as the `FM_BITCOIND_RPC_ENV`, would be appropriate?

https://github.com/fedimint/fedimint/blob/c5cb7dc811ac036cb976d4481471df3ec7491a7f/fedimint-bitcoind/src/bitcoincore_rpc.rs#L33-L59

**Comment by anmode** [comment]
I think we have to create a cookie file. isn't?                                                                                                                                       
```# Create the cookie file
echo "rpcuser=bitcoin" > $FM_BTC_DIR/.cookie
echo "rpcpassword=bitcoin" >> $FM_BTC_DIR/.cookie
chmod 400 $FM_BTC_DIR/.cookie`

**Comment by anmode** [comment]
Cookie based authentication is used when no rpc password is provided. The 0.12 release of Bitcoin Core had the following to say about it:                                                                                                                                                                                   
 ```When no -rpcpassword is specified, the daemon now uses a special ‘cookie’ file for authentication. This file is generated with random content when the daemon starts, and deleted when it exits. Its contents are used as authentication token. Read access to this file controls who can access through RPC. By default it is stored in the data directory but its location can be overridden with the option -rpccookiefile.```     
                                                                                                                                                                                                                   but here ig we have to create our cookie file like this-                                                                                                                                                                             
 ```# Create the cookie file
echo "rpcuser=bitcoin" > $FM_BTC_DIR/.cookie
echo "rpcpassword=bitcoin" >> $FM_BTC_DIR/.cookie
chmod 400 $FM_BTC_DIR/.cookie

**Comment by elsirion** [comment]
> I guess the problem would be how to get the path for the cookie file, a new environment variable as the `FM_BITCOIND_RPC_ENV`, would be appropriate?

Yes, I think that's the way to go here. If you are interested you could also look into #1994 (related but not a blocker, don't if it feels too involved to touch the `ServerModule`/`IServerModule` interfaces, we need better docs on these).

**Comment by elsirion** [comment]
@VeronicaTee Assigned the issue to you.
